lib/utils/util.js

```diff
@@ -478,3 +478,256 @@ util.storeOneInCache = function (processedOutput, cache) {
 
   this.storeInCache(processedOutput, cache, placer);
 };
+
+/**
+ * @typedef {import('../wallet/wallet)} Wallet
+ */
+
+/**
+ * Creates a backup from provided wallet
+ * @param {Wallet} wallet wallet
+ * @returns {Promise<WalletBackup>} WalletBackup
+ */
+util.backupWallet = async function (wallet) {
+  const unlock1 = await wallet.writeLock.lock();
+  const unlock2 = await wallet.fundLock.lock();
+
+  try {
+    const accountsBackup = [];
+    const accountNames = await wallet.getAccounts();
+
+    const masterKey = wallet.master;
+    const masterKeyData = masterKey.writeKey().toString('hex');
+
+    for (const accountName of accountNames) {
+      const account = await wallet.getAccount(accountName);
+      accountsBackup.push(await util.backupAccount(account));
+    }
+
+    const blinds = await wallet.getBlinds();
+
+    return {
+      ...wallet.getJSON(),
+      blinds,
+      masterKeyData: masterKeyData,
+      accounts: accountsBackup
+    };
+  } finally {
+    unlock2();
+    unlock1();
+  }
+};
+
+/**
+ * @typedef {import('../wallet/account')} Account
+ */
+
+/**
+ * Creates a Backup from provided account object
+ * @param {Account} account ...
+ * @returns {Promise<AccountBackup>} AccountBackup
+ */
+util.backupAccount = async function (account) {
+  const {
```

Comment:
For watch only wallets, when we take the backup we store the accountKey (xpub key of relevant account) 
within the backup file as well. Below backup file contains 2 accounts for a watchonly wallet.


```
{
        "wallet":
        {
            "network": "regtest",
            "wid": 1,
            "id": "realwalletwatcher",
            "watchOnly": true,
            "accountDepth": 2,
            "token": "59c41f247c6506e7ebd3a2a1a8a19c9546fc57599de5ceaf70d29e257b037a29",
            "tokenDepth": 0,
            "master":
            {
                "encrypted": false
            },
            "balance": null
        },
        "blinds":
        [],
        "masterKeyData": "3e3b7ff8c118ac7f0166b8b0a600e5fefb614f2e437203555b3aa7bdba6df61235b9bfde16c7e5731b3143ce29e5518b4715509cfe13d5c8c264d1dbb7ee90520100010210fee966279f19346f2be0d5f652f7b7cbfef5fdd829100f1a3d2684bc710ddb",
        "accountKey": "rpubKBBmaxiEgeEsAsStTPLd5egPwL5Q38Kg9XFw2qQ4kveyL76D7ZeniF9K3iEwk71h2HsvsjT5PnNCbybckV4yUMrMd2bvu8tDBrKGwV6JrhcG",
        "accounts":
        [
            {
                "name": "default",
                "type": 0,
                "m": 1,
                "n": 1,
                "accountKey": "rpubKBBmaxiEgeEsAsStTPLd5egPwL5Q38Kg9XFw2qQ4kveyL76D7ZeniF9K3iEwk71h2HsvsjT5PnNCbybckV4yUMrMd2bvu8tDBrKGwV6JrhcG",
                "keys":
                [],
                "receiveDepth": 1,
                "changeDepth": 1
            },
            {
                "name": "account1",
                "type": 0,
                "m": 1,
                "n": 1,
                "accountKey": "rpubKBBmaxiEgeEsAuvuy5H56EBAKHMjca87AyUoJRTq7zraUird8Tp9Pz3VJtKbWdgqJQkeqgJfBioky6ZAfPAk3oQk3VtNKSM42vKcPtq8VDhB",
                "keys":
                [],
                "receiveDepth": 1,
                "changeDepth": 1
            }
        ]
}
```


And when we restore the wallet the sequence goes as below:

Wallet gets created first and a default (with name 'default') account, if the wallet is watchOnly, it looks for xpubKey, if it 
does exist it completes creation of wallet and the default account.
Restore flow proceeds restoring the accounts if watchOnly wallet has more than one account, and since 
we do restore using the 

```
await wallet.ensureAccount({
    name,
    type,
    m,
    n,
    accountKey,
    keys
  });
```

  if account exists (the default account will) it wont attempt to create it, if not, a new watchOnly account
  will be created again using the provided accountKey associated with the account.

  During initial creation watchOnly wallets will require the xpubkey of the default account,

```
  curl --location --request PUT 'http://localhost:14039/wallet/realwalletwatcher' \
--header 'Authorization: Basic OnJvbw==' \
--header 'Content-Type: application/json' \
--header 'Cookie: ajs_user_id=%2293536597-dc7a-44fb-95bf-a2140af8f58e%22' \
--data-raw '{
    "watchOnly": true,
    "accountKey": "rpubKBBmaxiEgeEsAsStTPLd5egPwL5Q38Kg9XFw2qQ4kveyL76D7ZeniF9K3iEwk71h2HsvsjT5PnNCbybckV4yUMrMd2bvu8tDBrKGwV6JrhcG"

}'
```
  

  And for every consecutive account creationg under the watchOnly wallet we will be required to provide a xpub key

```
  curl --location --request PUT 'http://localhost:14039/wallet/realwalletwatcher/account/account1' \
--header 'Authorization: Basic OnJvbw==' \
--header 'Content-Type: application/json' \
--header 'Cookie: ajs_user_id=%2293536597-dc7a-44fb-95bf-a2140af8f58e%22' \
--data-raw '{
    "watchOnly": true,
    "accountKey": "rpubKBBmaxiEgeEsAuvuy5H56EBAKHMjca87AyUoJRTq7zraUird8Tp9Pz3VJtKbWdgqJQkeqgJfBioky6ZAfPAk3oQk3VtNKSM42vKcPtq8VDhB"
}'
```

So the restore flow is analogous to the creation flow.